### ISSUE TITLE
Finalize design for Composer header

### ISSUE THREAD

**Issue by glynnis** [MapStory/mapstory#275]
This story depends on #273.
- [x] Designing for a story that has been published, but has additional changes in draft mode but needs to be "republished" to push those changes to their live story

**Comment by glynnis** [comment]
Possibly helpful for terminology: public vs. private, changing the meaning and usage of the word "draft"

**Comment by glynnis** [comment]
This story will be informed by the TEM regarding Story Saving, but I can do some initial sketches before then.

**Comment by glynnis** [comment]
Related: https://github.com/MapStory/mapstory/issues/725

**Comment by glynnis** [comment]
Some notes that I've validated through a conversation with @sarasafavi:

Three story states can be communicated through the UI:
* Draft (not currently published)
* Published/public, with no staged changes
* Published/public, with staged changes that have not been published yet
  * On the backend: make a duplicate of the published one, where the duplicate has a "draft" flag and is not published. When we're ready to publish changes, the duplicate record with the `draft=true` gets changed to `draft=false` (and overwrite the record)
  * From this state, a user can revert to the public version that has no staged changes. If they're working on a story that's an original draft, their only option would be to discard the entire draft (since there's no public version to revert to).
  * If there's a public version of a draft that the user is working on, that's an important state to communicate (i.e. they have staged changes they haven't published yet).

**Comment by glynnis** [comment]
#725 is related and covers the technical aspects that affect this design story. I posted this diagram there, but posting here too since it'll help with design:

![04-06-draftartboard 1](https://cloud.githubusercontent.com/assets/1529366/26454455/ca58ae00-412c-11e7-991e-3ac49593db49.png)
^ Possible states and flows for a story

**Comment by glynnis** [comment]
I confirmed with @davisc that Preview Mode will be part of MVP.

**Comment by glynnis** [comment]
I still need to review these with the team so they aren't final yet, but here are the different states I have designed for w/r/t the header:

## Autosave Text
<img width="2160" alt="composer - header - autosave text" src="https://cloud.githubusercontent.com/assets/1529366/26596199/88f633f0-4533-11e7-8b71-f046820cd28d.png">

## Help Text
<img width="2224" alt="composer - header - help text" src="https://cloud.githubusercontent.com/assets/1529366/26596200/88f82340-4533-11e7-8467-3b35288e888d.png">

## MapStory Title
<img width="2180" alt="composer - header - title" src="https://cloud.githubusercontent.com/assets/1529366/26596201/88faa836-4533-11e7-9fad-d8bdcbe388ad.png">

**Comment by glynnis** [comment]
## Button states
<img width="2528" alt="composer - header - buttons" src="https://cloud.githubusercontent.com/assets/1529366/26596766/71e44b78-4535-11e7-80bd-ce7864afe10d.png">

**Comment by glynnis** [comment]
Calling this one done!

**Comment by glynnis** [comment]
You can also find these screens on InVision here: https://invis.io/PKBXSE1RJ
